### ISSUE TITLE
Recent App Store Rejection, for Non-public API usage

### ISSUE THREAD

**Issue by fflyer09** [transistorsoft/cordova-background-geolocation-lt#959]
Title: Recent App Store Rejection for Non-public API usage
## My Environment
* Plugin version:
	cordova-background-geolocation-lt 3.0.8
* Platform: iOS or Android
	iOS
* OS version:
	10 and up
* Device manufacturer / model:
	all iPhones
* Cordova version (`cordova -v`):
	8.0.0
* Cordova platform version (`cordova platform ls`):
	Installed platforms:
	  ios 4.5.5
* Plugin config provided to `#ready`:
        debug: false,
        desiredAccuracy: 0,
        distanceFilter: 10,
        useSignificantChangesOnly:true,
        locationAuthorizationRequest:"WhenInUse"

##Context of the problem ...
My app was approved back in October 2018, have gone through multiple versions, submissions through Xcode to update the app. Last successsfully pushed update was in June, 2019, went to push an update yesterday with NO changes to any libraries or components, just CSS tweaking and I received an Apple App Store rejection because of Non-public API usage. Two calls were in cordova-background-geolocation-lt component, the last call was in cordova-plugin-cocoalumberjack. The offending calls were:

1). databasePool:didAddDatabase:
2). databasePool:shouldAddDatabaseToPool:
3). ddSetLogLevel:

I searched the project directory for each of the calls listed above and here is the output.

$ grep -rl 'databasePool:didAddDatabase:' .
	./plugins/cordova-background-geolocation-lt/src/ios/TSLocationManager.framework/TSLocationManager
	./node_modules/cordova-background-geolocation-lt/src/ios/TSLocationManager.framework/TSLocationManager

$ grep -rl 'databasePool:shouldAddDatabaseToPool:' .
	./plugins/cordova-background-geolocation-lt/src/ios/TSLocationManager.framework/TSLocationManager
	./node_modules/cordova-background-geolocation-lt/src/ios/TSLocationManager.framework/TSLocationManager

$ grep -rl 'ddSetLogLevel:' .
	./plugins/cordova-plugin-cocoalumberjack/src/ios/CocoaLumberjack.framework/Headers/DDLog.h
	./plugins/cordova-plugin-cocoalumberjack/src/ios/libCocoaLumberjack.a
	./node_modules/cordova-plugin-cocoalumberjack/src/ios/CocoaLumberjack.framework/Headers/DDLog.h
	./node_modules/cordova-plugin-cocoalumberjack/src/ios/libCocoaLumberjack.a

Haven't received a notice like this before from Apple, surprised to see it now after passing review and having done many updates through Xcode since.

**Comment by christocracy** [comment]
It’s a false positive from [fmdb](https://github.com/ccgus/fmdb/issues/739).  Are you sure it’s not a *warning*?

**Comment by fflyer09** [comment]
Was able to upload through XCODE the first time. Usually it just takes a few minutes for me to receive notification that the update was accepted and I can immediately proceed to push out to test groups. This time I received the following email below and can not proceed further in App Store Connect.

Dear Developer,

We identified one or more issues with a recent delivery for your app, <MY APP>. Please correct the following issues, then upload again. The '_setAlwaysRunsAtForegroundPriority:' one related to the Cordova web view. I didn't include that in my original post above. I have notice another on 

ITMS-90338: Non-public API usage - The app references non-public selectors in Mingoal: _setAlwaysRunsAtForegroundPriority:, databasePool:didAddDatabase:, databasePool:shouldAddDatabaseToPool:, ddSetLogLevel:. If method names in your source code match the private Apple APIs listed above, altering your method names will help prevent this app from being flagged in future submissions. In addition, note that one or more of the above APIs may be located in a static library that was included with your app. If so, they must be removed. If you think this message was sent in error and that you have only used Apple-published APIs in accordance with the guidelines, send the app's Apple ID, along with detailed information about why you believe the above APIs were incorrectly flagged, to appreview@apple.com. For further information, visit the Technical Support Information at http://developer.apple.com/support/technical/

Best regards,

The App Store Team

**Comment by fflyer09** [comment]
Also I omitted the reference to _setAlwaysRunsAtForegroundPriority: in my original post because it involves Cordova web view

**Comment by christocracy** [comment]
`3.0.8` completely removed the unused `DatabasePool` class.  Are you sure you're submitting with `3.0.8`?

I suggest you fix your `_setAlwaysRunsAtForegroundPriority` issue and re-submit.

**Comment by fflyer09** [comment]
I did make a mistake, I was on version 2.12.2 when it failed upload to apple the first time. It worked all the way through to Jun 22nd 2019, but failed yesterday. The grep strings I provided above were taken from a the project directory with the 2.12.2 component replaced with the 3.0.8 component. 

After updating the geo component to 3.0.8 and ionic web view I was able to submit successfully as you suggested. I do receive warnings now for the same API functions listed above which I have never seen before yesterday, but it does go through now. I'm not receiving warnings for '_setAlwaysRunsAtForegroundPriority:' ... but still for 'databasePool:didAddDatabase:', 'databasePool:shouldAddDatabaseToPool:', 'ddSetLogLevel:'
